data_buffer.cpp

```diff
@@ -452,17 +452,19 @@ Vector2 DataBuffer::read_vector2(CompressionLevel p_compression_level) {
 }
 
 Vector2 DataBuffer::add_normalized_vector2(Vector2 p_input, CompressionLevel p_compression_level) {
+	const uint32_t is_not_zero = p_input.length_squared() > CMP_EPSILON;
+
 #ifdef DEBUG_ENABLED
-	ERR_FAIL_COND_V(p_input.is_normalized() == false, p_input);
+	ERR_FAIL_COND_V_MSG(p_input.is_normalized() == false && is_not_zero, p_input, "[FATAL] The encoding failed because this function expects a normalized vector.");
```

Comment:
is this correct? Unless I'm not reading it wrongly it's contradictory, like it will not complain when:
- input is normalized 
- and at the save time input length is zero

Reply:
It will stop the execution if the input is not normalized and it's not 0.